__tests__/server.test.js

```diff
@@ -35,4 +37,63 @@ describe("GET /api", ()=>{
             expect(endpoints).toEqual(endpoints)
         })
     })
+})
+
+describe("GET /api/articles/:article_id", ()=>{
+    test("200: Received a Response", ()=>{
+        return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({body: {article}})=>{
+            expect(article).toMatchObject({
+                article_id: 1,
+                author: "butter_bridge",
+                title: "Living in the shadow of a great man",
+                body: "I find this existence challenging",
+                topic: "mitch",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 100,
+                article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+            })
+        })
+    })
+    test("404: Resource not found based on a valid integer but no record", ()=>{
+        return request(app)
+        .get('/api/articles/999')
+        .expect(404)
+        .then(({body})=>{
+            expect(body.msg).toBe("Article not found")
+        })
+    })
+    test("400: Invalid data type", ()=>{
+        return request(app)
+        .get("/api/articles/banana")
+        .expect(400)
+        .then(({body})=>{
+            expect(body.msg).toBe("Bad request")
+        })
+    })
+})
+
+describe("GET /api/articles", ()=>{
+    test("200 - Gets all articles with comment count", ()=>{
+        return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({body: {articles}})=>{
+            expect(articles).toBeSortedBy("created_at", {descending: true})
+            articles.forEach(article => {

```

Comment:
In order to do a forEach you should be asserting the length of the returned body.